lib/config_files/base_model.rb

```diff
@@ -26,11 +27,41 @@ def save(changes_only: false)
 
     def load
       self.data = @parser.parse(@file_handler.read(@file_path))
+      @loaded = true
+    end
+
+    # powerfull method that sets any value in grub config. It try to be
+    # smart to at first modify existing value, then replace commented out code
+    # and if even that doesn't work, then append it at the end
+    # @note prefer to use specialized methods of children
+    def generic_set(key, value)
+      modify(key, value) || uncomment(key, value) || add_new(key, value)
+    end
+
+    # powerfull method that gets unformatted any value in grub config.
+    # @note prefer to use specialized methods of children
+    def generic_get(key)
+      data[key]
     end
 
   protected
 
+    # TODO: move it to generic place together with generic set and get
+    def self.attributes(attrs)
+      attrs.each_pair do |key, value|
+        define_method(key) do
+          generic_get(value)
+        end
+
+        define_method(:"#{key.to_s}=") do |target|
+          generic_set(value, target)
+        end
+      end
+    end
+    private_class_method :attributes
```

Comment:
I'm not sure if ```attributes``` is a good name. It sounds nicely declarative, but I can easily foresee a future in which we add an introspection method to ask for the known attributes like this:
```
puts "ConfigFiles::Grub knows about the following attributes #{ConfigFiles::Grub.attributes}"
```